Assets/Mirror/Runtime/NetworkServer.cs

```diff
@@ -877,15 +877,29 @@ static void OnClientReadyMessage(NetworkConnection conn, ReadyMessage msg)
 
         // default remove player handler
         static void OnRemovePlayerMessage(NetworkConnection conn, RemovePlayerMessage msg)
+        {
+            RemovePlayerForConnection(conn, true);
+        }
+
+        /// <summary>
+        /// Removes the player object from the connection
+        /// </summary>
+        /// <param name="conn">The connection of the client to remove from</param>
+        /// <param name="destroyServerObject">Indicates whether the server object should be destroyed</param>
+        public static void RemovePlayerForConnection(NetworkConnection conn, bool destroyServerObject)
         {
             if (conn.identity != null)
             {
-                Destroy(conn.identity.gameObject);
+                if (destroyServerObject)
+                    Destroy(conn.identity.gameObject);
+                else
+                    UnSpawn(conn.identity.gameObject);
+
                 conn.identity = null;
             }
             else
             {
-                Debug.LogError("Received remove player message but connection has no player");
+                if (LogFilter.Debug) Debug.Log($"Connection {conn} has no identity");
```

Comment:
Why is this no longer an error? If someone is trying to remove a null player shouldn't they get an error?

Reply:
For the same reason we no longer throw an error for removing authority when none was assigned.

Reply:
I Don't know what the reason for that was. Can you link or repost it here?

Reply:
Two errors removed in #1113 as excessively noisy.  One was from the null check...we just ignore if no owner because the end result is the same...no owner, and prevents double calls from generating unnecessary noise.  The other was eliminated through simplification that we no longer care who the current owner is...if there is one at all we remove it.

In this case, if there's no player to remove, call it good and shut up about it.